osu.Game/Overlays/Direct/DirectPanel.cs

```diff
@@ -40,6 +38,38 @@ public abstract class DirectPanel : Container
         private BeatmapManager beatmaps;
         private NotificationOverlay notifications;
 
+        private DownloadBeatmapSetRequest downloadRequest;
+        protected DownloadBeatmapSetRequest DownloadRequest
+        {
+            get { return downloadRequest; }
+            set
+            {
+                if (value == null || downloadRequest == value) return;

```

Comment:
rather than early returning on null, you should probably properly support null or just enforce non-null (and do a null check in the ctor code instead). we're trying in general to avoid the scenario where setting a property then getting that same property returns a different result.